File: content/courses/parallel-computing-introduction/codes/mpi1.cxx
#include <iostream>
#include "mpi.h"

using namespace std;

int main(int argc, char *argv[]) {
    int rank, npes;

    MPI_Init(&argc, &argv);
    MPI_Comm_size(MPI_COMM_WORLD, &npes);
    MPI_Comm_rank(MPI_COMM_WORLD, &rank);

    if ( rank == 0 ) {
        cout<<"Running on "<<npes<<"  Processes\n";
    }
    cout<<"Greetings from rank "<<rank<<"\n";

    MPI_Finalize();

}

File: content/courses/parallel-computing-introduction/codes/mpi1.f90
program hello
use mpi

    integer :: myrank, nprocs
    integer :: err

    call MPI_INIT(err)
    call MPI_COMM_RANK(MPI_COMM_WORLD, myrank, err)
    call MPI_COMM_SIZE(MPI_COMM_WORLD, nprocs, err)

    if ( myrank == 0 ) then
        print *, 'Running on ',nprocs,' Processes'
    endif

    print *, 'Greetings from process ', myrank
    call MPI_FINALIZE(err)


end program
